src/main/java/com/github/rholder/retry/WaitStrategies.java

```diff
@@ -310,6 +394,40 @@ public long computeSleepTime(Attempt failedAttempt) {
     }
 
     @Immutable
+    private static final class RandomExponentialWaitStrategy implements WaitStrategy {
+        private static final Random RANDOM = new Random();
+        private final long multiplier;
+        private final long minimumWait;
+        private final long maximumWait;
+
+        public RandomExponentialWaitStrategy(long multiplier,
+                                             long minimumWait,
+                                             long maximumWait) {
+            Preconditions.checkArgument(multiplier > 0L, "multiplier must be > 0 but is %d", multiplier);
+            Preconditions.checkArgument(minimumWait >= 0L, "minimumWait must be >= 0 but is %d", minimumWait);
+            Preconditions.checkArgument(maximumWait > minimumWait, "maximumWait must be > minimumWait but is %d", maximumWait);
+            Preconditions.checkArgument(multiplier < maximumWait, "multiplier must be < maximumWait but is %d", multiplier);
+            this.multiplier = multiplier;
+            this.minimumWait = minimumWait;
+            this.maximumWait = maximumWait;
+        }
+
+        @Override
+        public long computeSleepTime(Attempt failedAttempt) {
+            long upperBound = Math.round(Math.pow(2, failedAttempt.getAttemptNumber()));
+            long time = Math.abs(RANDOM.nextLong()) % upperBound;
+            long result = Math.round(multiplier * time);
+            if (result < minimumWait) {
+                result = minimumWait;
+            }
+            if (result > maximumWait) {
+                result = maximumWait;
+            }
+            return result >= 0L ? result : 0L;
```

Comment:
I left it this way for consistency with the other code in this file.  Is it worth it to replace them all to be consistent?